### ISSUE TITLE
Torchserve 0.8.1: custom metrics are not caught correctly

### ISSUE THREAD

**Issue by EgonFerri** [pytorch/serve#2450]
### 🐛 Describe the bug

I am trying to collect some custom metrics in a custom handler. 

To grant reproducibility, I used [alexnet tutorial](https://github.com/pytorch/serve/tree/86d440041b663961c71a6262fe648111d85b27d8/examples/image_classifier/alexnet)
with this minimal custom handler:

```python

import os
import time
import torch
from torchvision import transforms
import torch.nn.functional as F
import torch
from ts.torch_handler.vision_handler import VisionHandler
from ts.utils.util  import map_class_to_label


class image_custom_handler(VisionHandler):

    topk = 2
    # These are the standard Imagenet dimensions
    # and statistics
    image_processing = transforms.Compose([
        transforms.Resize(256),
        transforms.CenterCrop(224),
        transforms.ToTensor(),
        transforms.Normalize(mean=[0.485, 0.456, 0.406],
                             std=[0.229, 0.224, 0.225])
    ])

    def set_max_result_classes(self, topk):
        self.topk = topk

    def get_max_result_classes(self):
        return self.topk

    def postprocess(self, data):
        metrics = self.context.metrics
        start_time = time.time()
        
        ps = F.softmax(data, dim=1)
        probs, classes = torch.topk(ps, self.topk, dim=1)
        probs = probs.tolist()
        classes = classes.tolist()
        
        stop_time = time.time()
        metrics.add_time("CustomMetric", 
                         round((stop_time - start_time) * 1000, 2),
                         None, "ms")
        return map_class_to_label(probs, self.mapping, classes)

```

so, when I launch the service, I would expect something like:

```stdout

alexnet   | 2023-07-07T14:26:48,107 [INFO ] W-spaces_1.0-stdout MODEL_METRICS - CustomMetric.Milliseconds:520.099|#Level:Error|#hostname:32c8472cc112,timestamp:1688740008

```

instead, I get:

```stdout

alexnet       | 2023-07-07T15:56:55,580 [INFO ] W-9000-alexnet_1.0-stdout org.pytorch.serve.wlm.WorkerLifeCycle - result=[METRICS]CustomMetric.Milliseconds:0.27|#ModelName:alexnet,Level:Model|#hostname:643dd69cbd72,1688745415,239ba013-0e24-49bf-944d-9159c8fac8b6, pattern=[METRICS]

```

furthermore, since the metric is missed, is not saved into logs/model_metrics.log

Note that until I used the 0.7.1 version, everything worked as expected. 

### Error logs

```stdout

alexnet       | 2023-07-07T15:56:55,580 [INFO ] W-9000-alexnet_1.0-stdout org.pytorch.serve.wlm.WorkerLifeCycle - result=[METRICS]CustomMetric.Milliseconds:0.27|#ModelName:alexnet,Level:Model|#hostname:643dd69cbd72,1688745415,239ba013-0e24-49bf-944d-9159c8fac8b6, pattern=[METRICS]

```

### Installation instructions

pip install torchserve torch-model-archiver torch-workflow-archiver

### Model Packaing

`torch-model-archiver --model-name alexnet --version 1.0 --model-file ./serve/examples/image_classifier/alexnet/model.py --serialized-file alexnet-owt-7be5be79.pth --handler image_custom_handler.py --extra-files ./serve/examples/image_classifier/index_to_name.json --export-path ./model_store -f `

### config.properties

```
inference_address=http://0.0.0.0:8080
management_address=http://0.0.0.0:8081
metrics_address=http://0.0.0.0:8082
disable_system_metrics='false'

models={\
  "spaces": {\
    "1.0": {\
        "defaultVersion": true,\
        "marName": "spaces.mar",\
        "minWorkers": 1,\
        "maxWorkers": 4,\
        "batchSize": 1,\
        "maxBatchDelay": 1000,\
        "responseTimeout": 120\
    }\
  }\
}
```

### Versions

torch                                    2.0.1
torch-model-archiver                     0.8.1
torch-tb-profiler                        0.4.1
torch-xla                                2.0
torchserve                               0.8.1
torchvision                              0.15.2

docker image:  pytorch/torchserve 
version: 0.8.1

### Repro instructions

```bash
git clone https://github.com/pytorch/serve.git

wget https://download.pytorch.org/models/alexnet-owt-7be5be79.pth

torch-model-archiver --model-name alexnet --version 1.0 --model-file ./serve/examples/image_classifier/alexnet/model.py --serialized-file alexnet-owt-7be5be79.pth --handler image_custom_handler.py --extra-files ./serve/examples/image_classifier/index_to_name.json --export-path ./model_store -f 

docker-compose -f docker-compose.yaml up
curl http://127.0.0.1:8090/predictions/alexnet -T ./serve/examples/image_classifier/kitten.jpg

```

where docker compose is just:

```docker
version: '3'

services:
  torchserve:
    image: pytorch/torchserve:0.8.1-cpu
    ports:
      - 8090:8080
      - 8091:8081
      - 8092:8082
    container_name: alexnet
    volumes: 
        - './model_store:/home/model-server/model-store'
    command:
      - 'torchserve --start'
      - '--model-store model-store'
      - '--models alexnet=alexnet.mar'
      - '--ts-config model-store/config.properties'
```

### Possible Solution

i think there is something that broke in log parsing in 0.8.1 (since everything was fine with 0.7.1), but I'm not sure.

thanks in advance to anyone willing to help.

**Comment by namannandan** [comment]
@EgonFerri starting `v0.8.0`, metrics started using a centralized configuration file to define the metrics that Torchserve will emit, this is shown in the documentation [here](https://github.com/pytorch/serve/blob/master/docs/metrics.md#central-metrics-yaml-file-definition). You can use the [default metrics configuration file](https://github.com/pytorch/serve/blob/master/ts/configs/metrics.yaml) as a starting point and add your custom metric `CustomMetric` under `model_metrics` as follows:
```
model_metrics:
  gauge:
    - name: CustomMetric
      unit: ms
      dimensions: [*model_name, *level]
```
Then, specify the custom metrics configuration file in `config.properties` as shown [here](https://github.com/pytorch/serve/blob/master/docs/metrics.md#user-manual---starting-torchserve-with-a-yaml-file-specified) and start torchserve.
This should emit `CustomMetric` to `model_metrics.log`

**Comment by EgonFerri** [comment]
OK, clear. Thanks for the quick response, today I'll fix my code.
I had seen the new metrics docs but it wasn't super clear to me if it was mandatory or alternative, since in the 0.8 release notes no breaking changes were declared for the metric system.
